frontend/src/ches/MergeForm.jsx

```diff
@@ -87,18 +108,27 @@ class MergeForm extends Component {
     let form = this.state.form;
     form.sender = event.target.value;
     this.setState({form: form, info: ''});
+    let preview = this.state.preview;
+    preview.allowed = this.canPreview();
+    this.setState({preview: preview});
   }
 
   onChangeSubject(event) {
     let form = this.state.form;
     form.subject = event.target.value;
     this.setState({form: form, info: ''});
+    let preview = this.state.preview;
+    preview.allowed = this.canPreview();
+    this.setState({preview: preview});
   }
 
   onChangePlainText(event) {
     let form = this.state.form;
     form.plainText = event.target.value;
     this.setState({form: form, info: ''});
+    let preview = this.state.preview;
```

Comment:
I think a bunch of these variables can probably be set as const due to lack of reassignment before end of scope.

Reply:
addressed in next commit.